vizro-ai/tests/integration/test_dashboard.py

```diff
@@ -17,14 +18,161 @@
+                controls_types = [1 if cntrls.count(controls_type["type"]) == controls_type["num"] else 0][0]
+            except IndexError:
+                controls_types = 0
+            controls_types_names.append(controls_types)
+            controls_types_names_report.append(
+                f'{controls_type["num"]} controls_type(s) {controls_type["type"]} '
+                f'for page {controls_type["page_num"]} is {bool(controls)}'
+            )
+
+    prescore = []
+    pages_exist.extend(pages_num)
+    prescore.extend(pages_exist)
+    prescore.extend(components_num)
+    prescore.extend(controls_num)
+    prescore.extend(components_types_names)
+    prescore.extend(controls_types_names)
+    score = sum(prescore)
+
+    pages_score = sum(pages_exist) / len(pages_exist)
+    components_score = sum(components_num) / len(components_num)
+    component_types_score = sum(components_types_names) / len(components_types_names)
+    controls_score = sum(controls_num) / len(controls_num)
+    controls_types_score = sum(controls_types_names) / len(controls_types_names)
+    total = [pages_score, components_score, component_types_score, controls_score, controls_types_score]
+    total_score = sum(total) / len(total)
+
+    with open(f"{report_dir}/report_model_{model_name}_{vizro_type}.csv", "a", newline="") as csvfile:
+        writer = csv.writer(csvfile, delimiter=",")
+        writer.writerow(
+            [
+                "timestamp",
+                "vizro_type",
+                "branch",
+                "python_version",
+                "model",
+                "prompt_tier",
+                "total_score",
+                "pages_score",
+                "components_score",
+                "component_types_score",
+                "controls_score",
+                "controls_types_score",
+            ]
+        )
+        writer.writerow(
+            [
+                datetime.now(),
+                vizro_type,
+                branch,
+                python_version,
+                model_name,
+                prompt_tier,
+                total_score,
+                pages_score,
+                components_score,
+                component_types_score,
+                controls_score,
+                controls_types_score,
+            ]
+        )
+
+    # for cmd output
+    print(f"Pages exists: {pages_exist_report}")  # noqa: T201
+    print(f"Correct pages number: {pages_num_report}")  # noqa: T201
+    print(f"Components: {components_num_report}")  # noqa: T201
+    print(f"Correct controls number: {controls_num_report}")  # noqa: T201
+    print(f"Correct components types: {components_types_names_report}")  # noqa: T201
+    print(f"Correct controls types: {controls_types_names_report}")  # noqa: T201
+    print(f"Total, {(score / len(prescore)):.4f}")  # noqa: T201
+
+
+@pytest.mark.easy_dashboard
 @pytest.mark.parametrize(
     "model_name",
     ["gpt-3.5-turbo"],
```

Comment:
Let's `gpt-4o` instead. `gpt-4o-mini` is recommended replacement for "gpt-3.5" already